sdk/search/azure-search-documents/src/main/java/com/azure/search/documents/models/ComplexSearchField.java

```diff
@@ -0,0 +1,52 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.search.documents.models;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+/**
+ * A helper Field model to build a simple search field.
+ */
+public class ComplexSearchField {
+    private String name;
+    private ComplexDataType dataType;
+    private List<Field> fieldList;
+
+    public String getName() {
+        return name;
+    }
+
+    public ComplexSearchField setName(String name) {
+        this.name = name;
+        return this;
+    }
+
+    public ComplexDataType getDataType() {
+        return dataType;
+    }
+
+    public ComplexSearchField setDataType(ComplexDataType dataType) {
+        this.dataType = dataType;
+        return this;
+    }
+
+    public List<Field> getFieldList() {
+        return fieldList;
+    }
+
+    public ComplexSearchField setSimpleFieldList(List<SimpleSearchField> fieldList) {
```

Comment:
Complex fields can also have complex fields. This method name doesn't work. I'd just keep the property `fields`, ie. `setFields`. That's what it is in track 1 and track 2 in .NET.